mem/idealmemcontroller/builder.go

```diff
@@ -0,0 +1,158 @@
+package idealmemcontroller
+
+import (
+	"github.com/sarchlab/akita/v3/mem/mem"
+	"github.com/sarchlab/akita/v3/pipelining"
+	"github.com/sarchlab/akita/v3/sim"
+)
+
+type Builder struct {
+	width             uint64
+	latency           int
+	maxNumTransaction int
+	freq              sim.Freq
+	capacity          uint64
+	engine            sim.Engine
+	clsize            int
+	topBufSize        int
+	storage           *mem.Storage
+
+	numCyclePerStage int
+	numStage         int
+	clPerCycle       int
+	pipeBufSize      int
+}
+
+// MakeBuilder returns a new Builder
+func MakeBuilder() Builder {
+	return Builder{
+		// width:             64,
+		latency:           100,
+		maxNumTransaction: 8,
+		freq:              1 * sim.GHz,
+		capacity:          4 * mem.GB,
+		clsize:            64,
```

Comment:
I think it is better to use CacheLine rather than CL. CL is easily confused with OpenCL.